From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Markus Kuhn <Markus.Kuhn@cl.cam.ac.uk>
Date: Fri, 28 Mar 2014 18:08:05 +0000
Subject: [PATCH] ensure appropriate error message with zero-length input file


diff --git a/pbmtools/jbgtopbm.c b/pbmtools/jbgtopbm.c
index a6a8c78..3a9a9d4 100644
--- a/pbmtools/jbgtopbm.c
+++ b/pbmtools/jbgtopbm.c
@@ -73,6 +73,7 @@ void read_file(unsigned char **buf, size_t *buflen, size_t *len, FILE *f)
       exit(1);
     }
   } while (!feof(f));
+  if (!*len) return;
   *buflen = *len;
   *buf = (unsigned char *) realloc(*buf, *buflen);
   if (!*buf) {
@@ -134,6 +135,7 @@ void diagnose_bie(FILE *fin)
   if (len < 20) {
     fprintf(f, "Error: Input file is %lu < 20 bytes long and therefore "
 	    "does not contain an intact BIE header!\n", (unsigned long) len);
+    free(bie);
     return;
   }
 
